tools/src/tensil/tools/compiler/OnnxFrontend.scala

```diff
@@ -111,21 +137,27 @@ class OnnxFrontend(
         dimensions = Vector(channelsOut, channelsIn, height, width)
       )
 
-    def apply(shape: Shape): MemoryDimensions =
-      if (shape.size == 1)
-        ConstsDimensions(shape(0))
-      else if (shape.size == 2)
-        ConstsDimensions(shape(0), shape(1))
-      else if (shape.size == 4)
-        ConstsDimensions(shape(0), shape(1), shape(2), shape(3))
-      else
+    def apply(shape: Shape, transpose: Boolean): MemoryDimensions =
+      if (transpose && shape.size != 2)
         throw new CompilerException(
-          s"Consts tensor shape of ${shape} is not supported"
+          s"Transposing consts is supported for 2D tensors only"
```

Comment:
Is this specified by ONNX or just a limitation of our current implementation?

Reply:
So far, I've only seen transpose applied to the arguments of `Gemm`, which are 2D tensors.